src/controllers/HighlightModeController.js

```diff
@@ -1,22 +1,75 @@
 import AnnotationModeController from './AnnotationModeController';
-import { THREAD_EVENT, CONTROLLER_EVENT } from '../constants';
+import { CLASS_ANNOTATION_LAYER_HIGHLIGHT, THREAD_EVENT, CONTROLLER_EVENT, CREATE_EVENT, TYPES } from '../constants';
+import CreateHighlightDialog from './CreateHighlightDialog';
 
 class HighlightModeController extends AnnotationModeController {
     /**
-     * Handles annotation thread events and emits them to the viewer
-     *
      * @inheritdoc
+     * @return {void}
+     */
+    setupSharedDialog() {
+        this.allowComment = !!(this.mode === TYPES.highlight_comment);
+        this.allowHighlight = !!(this.mode === TYPES.highlight);
+
+        this.createDialog = new CreateHighlightDialog(this.container, {
+            isMobile: this.isMobile,
+            hasTouch: this.hasTouch,
+            allowComment: this.allowComment,
+            allowHighlight: this.allowHighlight,
+            localized: this.localized
+        });
+        this.createDialog.createElement();
+
+        this.onDialogPendingComment = this.onDialogPendingComment.bind(this);
+        this.onDialogPost = this.onDialogPost.bind(this);
+        this.destroyPendingThreads = this.destroyPendingThreads.bind(this);
+
+        this.createDialog.addListener(CREATE_EVENT.init, () => this.emit(THREAD_EVENT.pending, this.mode));
+
+        if (this.allowComment) {
+            this.createDialog.addListener(CREATE_EVENT.comment, this.onDialogPendingComment);
+            this.createDialog.addListener(CREATE_EVENT.post, this.onDialogPost);
+        }
+
+        if (this.allowHighlight) {
+            this.createDialog.addListener(CREATE_EVENT.plain, this.onDialogPost);
+        }
+    }
+
+    /**
+     * @inheritdoc
+     * @return {void}
+     */
+    destroy() {
+        this.createDialog.removeListener(CREATE_EVENT.comment, this.onDialogPendingComment);
+        this.createDialog.removeListener(CREATE_EVENT.post, this.onDialogPost);
+        this.createDialog.removeListener(CREATE_EVENT.plain, this.onDialogPost);
+
+        this.createDialog = null;
+        super.destroy();
+    }
+
+    /**
+     * Notify listeners of post event and then clear the create dialog
+     *
      * @private
-     * @param {AnnotationThread} thread - The thread that emitted the event
-     * @param {Object} [data] - Annotation thread event data
-     * @param {string} [data.event] - Annotation thread event
-     * @param {string} [data.data] - Annotation thread event data
+     * @param {string} commentText Annotation comment text
+     * @return {void}
+     */
+    onDialogPendingComment() {
+        this.emit(CONTROLLER_EVENT.createPendingThread);
+    }
+
+    /**
+     * @inheritdoc
      * @return {void}
      */
     handleThreadEvents(thread, data) {
         switch (data.event) {
```

Comment:
Do you need to use a switch statement if you're only using one case? that could be for later cleanup, if you'd prefer